lsif/src/dump-manager/routes/uploads.ts

```diff
@@ -0,0 +1,100 @@
+import express from 'express'
+import { Logger } from 'winston'
+import { Span } from 'opentracing'
+import { wrap } from 'async-middleware'
+import { addTags, TracingContext, logAndTraceCall } from '../../shared/tracing'
+import { pipeline as _pipeline } from 'stream'
+import { promisify } from 'util'
+import * as fs from 'mz/fs'
+import * as settings from '../settings'
+import { dbFilename, uploadFilename } from '../../shared/paths'
+import { ThrottleGroup, Throttle } from 'stream-throttle'
+
+const pipeline = promisify(_pipeline)
+
+/**
+ * Create a router containing the upload endpoints.
+ *
```

Comment:
```suggestion
 * Create a router containing the upload endpoints.
 * The upload service exports the following methods:
 *   - `GET /<upload-id>/raw`: Streams the unprocessed content of an upload.
 *   - `POST /<upload-id>/raw`: Writes or replaces the unprocessed content of an upload.
 *   - `POST /<upload-id>`: Writes or replaces the processed content of an upload.
```

Do we have a definition somewhere of the formats of these streams? We don't seem to be doing any validation, so what would keep me from (say) posting a processed database to the `raw` endpoint or an unprocessed upload to the cooked endpoint?

Relatedly: I guess we probably already use these routes and can't change them, but I feel like we shouldn't have a "naked" `/<upload-id>` endpoint, but should explicitly distinguish the type of data, e.g., `/<upload-id>/db` or something like that. If we're already using it this can wait.

Reply:
These are new routes so we can do whatever we want! I just threw these together really fast to get a POC working, so suggestions for any API stuff welcome (also, all of the lsif-server stuff is internal and we are free to change it at any level we'd like, as long as we update the client in the go code).

Reply:
Also, these routes are _only_ used so that the lsif-server and dump processor can communicate with some shared state (SQLite indexes and raw upload files). So there's a very small vector for this going wrong (they're only referenced twice in code and aren't user endpoints).

Reply:
We also have the OpenAPI doc `docs/api.yaml` for the server that I plan to update after the API changes settle.

Reply:
Updated the route names a bit, will add openapi doc in part 4.